### PR TITLE
Multiple insert

### PR DESCRIPTION
**Pull request by neliades**
@mfosho 
Please review my recent progress... you can ignore the dummyDataNew file - it was/is used for testing and development.

### CODE REVIEW

File: database-mysql/dummyDataTenMillion.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+//======================================
+//estimated time to completion: 30.32 minutes
+//======================================
```
Comment (benmangold) [inline database-mysql/dummyDataTenMillion.js:3]: just popping in here, I really like this comment style.  super readable and breaks up the code really well

File: database-mysql/dummyDataNew.js [hunk 2]
```diff
@@ -339,57 +342,254 @@ function Data() {
...
+
+
+
+
+
+
+//===========================================
+var insertNumOfRows = (numOfRows, callback) => {
+  var all = [];
+  for (var i =0; i < numOfRows; i++) {
```
Comment (ghost) [inline database-mysql/dummyDataNew.js:410]: minor but space after = in var i = 0, can also do formatDocument command+shift+p, and type in format document in the search bar

File: database-mysql/dummyDataNew.js [hunk 3]
```diff
@@ -339,57 +342,254 @@ function Data() {
...
+    var data = new Data;
+    var arr = [JSON.stringify(data.name), JSON.stringify(data.roomType), JSON.stringify(data.roomTypeDetails), JSON.stringify(data.city), JSON.stringify(data.cityDetails), JSON.stringify(data.listingDetails), JSON.stringify(data.guestAccessDetails), JSON.stringify(data.interactionGuestsDetails), JSON.stringify(data.otherDetails)];
```
Comment (ghost) [inline database-mysql/dummyDataNew.js:412]: if using every key/value in the object, depending on speed, can use a for in loop for the object (assuming Data is an object) so do not need to repeat JSON.stringify

File: database-mysql/dummyDataNew.js [hunk 4]
```diff
@@ -339,57 +342,254 @@ function Data() {
...
+    if (seconds < 0) {
+      minutes -= 1;
+      seconds = 60 + seconds;
+    }
+    if (milliseconds < 0) {
+      seconds -= 1;
+      milliseconds = 999 + milliseconds;
+    }
+    
+    console.log('time elapsed:', minutes, 'm : ', seconds, 's : ', milliseconds, 'ms');
```
Comment (ghost) [inline database-mysql/dummyDataNew.js:579]: can use console.time('anything') and console.timeEnd('anything') to track run times

File: database-mysql/dummyDataNew.js [hunk 5]
```diff
@@ -339,57 +342,254 @@ function Data() {
...
-
+  
   promise.then(function() {
     wrapper();
   });
 }
 
-wrapper();
\ No newline at end of file
+wrapper();
```
Comment (ghost) [inline database-mysql/dummyDataNew.js:594]: can add a catch statement

File: database-mysql/dummyDataTenMillion.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+      function(resolve) {
+        insertNumOfRows(500, () => {
+            resolve();
+        });
+    });
+    //========
+    //recurses when promise resolves
+    promise.then(function() {
+      wrapper();
+    });
```
Comment (ghost) [inline database-mysql/dummyDataTenMillion.js:98]: another way to format this flow/a promise would be for the promise to generate the data first, then when it resolves, send the resolve() with the data (e.g. resolve(data)), and in the then statement insert into the database, and ultimately catch any errors

**Review by ghost — COMMENTED** [review]
Looks good, let me know if you have any questions about my comments!

**Pull request merged**